utilities/gitopsctl/implementations/download-job/download-job.go

```diff
@@ -0,0 +1,350 @@
+			url:  fileURL,
+			path: "./downloaded/" + filePath,
+		})
+
+	}
+
+	fmt.Println("* Downloading", len(entriesToDownload), "files")
+
+	downloadURLsMultithreaded(entriesToDownload)
+
+	fmt.Println("* Download complete")
+
+	return nil
+}
+
+// skipFileDownload filters out URLs from download: returns true if a URL should not be downloaded, false otherwise
+func skipFileDownload(url string) bool {
+
+	// remove everything before, and including, '/artifacts/'
+	{
+		artifactsStrConst := "/artifacts/"
+
+		index := strings.Index(url, artifactsStrConst)
+		if index == -1 {
+			return false
+		}
+
+		url = url[index+len("/artifacts/"):]
+	}
+
+	// remove everything before, and including, 'appstudio-e2e-tests'
+	{
+		e2eTestsConst := "appstudio-e2e-tests"
+
+		e2eTestsIndex := strings.Index(url, e2eTestsConst)
+		if e2eTestsIndex == -1 {
+			return false
+		}
+
+		url = url[e2eTestsIndex+len(e2eTestsConst):]
+	}
+
+	if skipTraverseURL(url) {
+		return true
+	}
+
+	if !(strings.Contains(url, "git") || strings.Contains(url, "appstudio")) {
+		return true
+	}
+
+	return false
+
+}
+
+// skipTraverseURL will return true if we should avoid traversing (recursing scanning) a given URL, false otherwise.
+func skipTraverseURL(url string) bool {
+
+	// various string constants where we need not bother traversing the URL
+	// - these are generally never going to be relevant to us in gitops service
+	toSkip := []string{
+		"jvm-build-service", "o11y", "quality-dashboard", "spi-system",
+		"rp_preproc", "kube-rbac-proxy", "haproxy", "pipelines-as-code",
+		"autoscaling", "enterprise-contract", "release-service", "tekton-results",
+		"toolchain-host-operator", "dora-metrics", "spi-demos-", "tekton-chains",
+		"spi-vault", "dev-sso", "kube-apiserver-proxy", "appstudio-grafana", "image-controller",
+		"kube-public", "kam-", "toolchain-member-operator", "metrics", "kube-apiserver",
+	}
+
+	for _, toSkipEntry := range toSkip {
+
+		if strings.Contains(url, toSkipEntry) {
+			return true
+		}
+
+	}
+
+	url = strings.ToLower(url)
+
+	if strings.Contains(url, "/namespaces/") {
+
+		if strings.Contains(url, "/namespaces/openshift-") && !strings.Contains(url, "git") {
+			return true
+		}
+
+		if !(strings.Contains(url, "git") || strings.Contains(url, "appstudio")) {
+			return true
+		}
+
+	}
+
+	return false
+}
+
+func extractAllFileURLsFromArtifactsPage(url string, depth int) ([]string, error) {
+
+	res := []string{}
+
+	data, err := downloadURLAsString(url)
+	if err != nil {
+		return res, fmt.Errorf("%v", err)
+	}
+
+	doc, err := html.Parse(strings.NewReader(data))
+	if err != nil {
+		return res, fmt.Errorf("%v", err)
+	}
+
+	urlSuffixConstant := "openshiftapps.com/"
+
+	host := url[:strings.Index(url, urlSuffixConstant)+len(urlSuffixConstant)-1]
+
+	links := extractLinks(doc, 0)
+	for index, link := range links {
+
+		if index == 0 { // The first link is always a reference to the parent
+			continue
+		}
+
+		if strings.Contains(link, "gsutil") {
```

Comment:
The link here points to another tool that downloads the files?  Have you tried it?

Reply:
I've added a comment to this line explaining what its purpose is.